pool/pool.go

```diff
@@ -898,14 +952,21 @@ func (p *pool) WaitForContainerPresence(ctx context.Context, cid *cid.ID, pollPa
 	defer ticker.Stop()
 	wdone := wctx.Done()
 	done := ctx.Done()
+
+	var cliPrm client.ContainerGetPrm
+
+	if cid != nil {
+		cliPrm.SetContainer(*cid)
+	}
```

Comment:
As soon as client library throws panics, I suppose it is okay to not check `nil` there and just go straight `cliPrm.SetContainer(*cid)`. It is invalid case when `cid` is nill.

The same comment applied for others `nil` check on mandatory arguments in pool package.

Reply:
W/o nil-check here program will panic with generic NPE, while `Client` panics with the exact reason. In future I hope we will accept `cid` by value, not a pointer.